internal/logic_handlers/organizationlogic/service_test.go

```diff
@@ -0,0 +1,674 @@
+			assertions: func(t *testing.T, paginated *model.PaginatedOrganizations, err error) {
+				require.Error(t, err, "unexpected get organizations error")
+				require.Contains(t, err.Error(), "get db:")
+			},
+		},
+	}
+}
+
+func TestService_GetOrganizations(t *testing.T) {
+	for _, tt := range getGetOrganizationTestCases() {
+		t.Run(tt.name, func(t *testing.T) {
+			_dependencies, cleanup := tt.getDependencies(t)
+			defer cleanup()
+
+			svc, err := New(&Config{
+				TxProvider: _dependencies.TxProvider,
+				Logger:     _dependencies.Logger,
+				Persistor:  _dependencies.Persistor,
+			})
+			require.NoError(t, err, "unexpected new error")
+
+			tt.mutations(t, _dependencies.Db)
+
+			paginatedOrganizations, err := svc.ListOrganization(tt.args.ctx, tt.args.filter)
+			tt.assertions(t, paginatedOrganizations, err)
+		})
+	}
+}
+
+type argsUpdateOrganizations struct {
+	ctx    context.Context
+	params *model.UpdateOrganization
+}
+
+type testCaseUpdateOrganization struct {
+	name            string
+	getDependencies func(t *testing.T) (*dependencies, func(ignoreErrors ...bool))
+	args            argsUpdateOrganizations
+	mutations       func(t *testing.T, db *sqlx.DB)
+	assertions      func(t *testing.T, params *model.UpdateOrganization, organization *model.Organization, err error)
+}
+
+func getTestCasesUpdateOrganizations() []testCaseUpdateOrganization {
+	return []testCaseUpdateOrganization{
+		{
+			name: "success",
+			args: argsUpdateOrganizations{
+				ctx: context.TODO(),
+				params: &model.UpdateOrganization{
+					Id: 1,
+					Name: null.String{
+						String: "Demby",
+						Valid:  true,
+					},
+					UserId: null.Int{
+						Int:   3,
+						Valid: true,
+					},
+				},
+			},
+			getDependencies: getConcreteDependencies,
+			mutations: func(t *testing.T, db *sqlx.DB) {
+				entry := mysqlmodel.Organization{
+					ID:            1,
+					Name:          "Demby",
+					CreatedBy:     null.IntFrom(3),
+					LastUpdatedBy: null.IntFrom(3),
+					CreatedAt:     time.Now(),
+					LastUpdatedAt: null.TimeFrom(time.Now()),
+					IsActive:      true,
+				}
+
+				err := entry.Insert(context.Background(), db, boil.Infer())
+				require.NoError(t, err, "error inserting sample data")
+			},
+			assertions: func(t *testing.T, params *model.UpdateOrganization, organization *model.Organization, err error) {
+				createdByConvToInt, _ := strconv.Atoi(organization.CreatedBy)
+				require.NoError(t, err, "unexpected error")
+				require.NotNil(t, organization, "unexpected nil organization")
+				assert.Equal(t, params.Id, organization.Id, "expected id to be equal")
+				assert.Equal(t, params.UserId.Int, createdByConvToInt, "expected created_by id to be equal")
+				assert.Equal(t, params.Name.String, organization.Name, "expected name to be equal")
+			},
+		},
+		{
+			name: "success-name-only",
+			args: argsUpdateOrganizations{
+				ctx: context.TODO(),
+				params: &model.UpdateOrganization{
+					Id: 1,
```

Comment:
no static

Reply:
Understood. Is it ok if we pass in args into the mutations from the test itself?